docs/source/data-access/data-dependency.rst

```diff
@@ -0,0 +1,87 @@
+
+For example, SWAPI L3 requires both SWAPI L2 files and MAG L1D (previously called L2pre) files. The SWAPI L2 job and the MAG L1D job are run independently, so there is no guarantee that they will finish
+at the same time. Let's assume that the MAG L1D job finishes first, since it is the lower level. When that file arrives, one of the downstream dependencies is going to be the SWAPI L3 processing.
+However, when batch starter checks the upstream dependencies for SWAPI L3, it will find that SWAPI L2 is missing. Therefore, processing won't start. Once the SWAPI L2 processing finishes,
+and the SWAPI L2 file arrives, the batch starter is triggered with that file. Once again, SWAPI L3 is a downstream dependency, but this time, both upstream dependencies for SWAPI L2 are present.
+Therefore, processing for SWAPI L3 can begin.
+
+The status of different files is recorded in the status tracking table. This table records the status of each anticipated output file as "in progress", "complete", or "failed." Through this,
+we can track processing for specific files and determine if a file exists quickly.
+
+Data Dependency Database Table
+------------------------------
+
+How does the SDC track which files are dependent on others? In order to decide what the downstream or upstream dependencies of a file are, and what the nature of those dependencies are, we
+need some way to request the upstream or downstream dependencies of a given file. The current dependencies between instruments are recorded in `Galaxy <https://lasp.colorado.edu/galaxy/display/IMAP/Dependency+Between+Instrument+-+Diagram>`_.
+
+We handle this using a SQL database. All of our databases are described in detail `here <https://lasp.colorado.edu/galaxy/display/IMAP/SDC+Database+Tables>`_.
+
+The database has the following structure:
+
+========== ===== ========== ===================== ================ ===================== ======================== ======================
+instrument level descriptor dependency_instrument dependency_level dependency_descriptor relationship_description DOWNSTREAM or UPSTREAM
+========== ===== ========== ===================== ================ ===================== ======================== ======================
+mag        l1a   norm-mago  mag                   l1b              norm-mago             HARD                     DOWNSTREAM
+mag        l1a   norm-magi  mag                   l1b              norm-magi             HARD                     DOWNSTREAM
+mag        l1d   norm       swapi                 l3               sci                   HARD                     DOWNSTREAM
+swapi      l2    sci        swapi                 l3               sci                   HARD                     DOWNSTREAM
+========== ===== ========== ===================== ================ ===================== ======================== ======================
```

Comment:
Can you add the upstream dependency you mentioned in the example above? I think it will help to see that in this table as well.